uqcsbot/remindme.py

```diff
@@ -0,0 +1,408 @@
+            user_id=reminder.user_id,
+            channel_id=reminder.channel_id,
+            time_created=reminder.time_created,
+            message=reminder.message,
+            time=reminder.time,
+            start_date=reminder.start_date,
+            end_date=reminder.end_date,
+            week_frequency=reminder.week_frequency
+        ))
+        db_session.commit()
+        db_session.close()
+
+    def _remove_reminder_from_db(self, reminder_id: int) -> Reminder:
+        """ Removes the reminder with id `reminder_id` from the Reminders table in the database """
+        db_session = self.bot.create_db_session()
+        reminder_query = db_session.query(Reminders).filter(Reminders.id == reminder_id).one()
+        removed_reminder = Reminder(
+            reminder_query.id,
+            reminder_query.user_id,
+            reminder_query.channel_id,
+            reminder_query.time_created,
+            reminder_query.message,
+            reminder_query.time,
+            reminder_query.start_date,
+            reminder_query.end_date,
+            reminder_query.week_frequency
+        )
+        db_session.delete(reminder_query)
+        db_session.commit()
+        db_session.close()
+
+        return removed_reminder
+
+    def _get_all_reminders(self) -> List[Reminder]:
+        """ Returns all the active reminders in the database in creation order """
+        db_session = self.bot.create_db_session()
+        reminders_query = db_session.query(Reminders)
+        db_session.close()
+
+        return [Reminder(
+            reminder.id,
+            reminder.user_id,
+            reminder.channel_id,
+            reminder.time_created,
+            reminder.message,
+            reminder.time,
+            reminder.start_date,
+            reminder.end_date,
+            reminder.week_frequency
+        ) for reminder in sorted(reminders_query, key=lambda reminder: reminder.time_created)]
+
+    def _get_user_reminders(self, user_id: int) -> List[Reminder]:
+        """ Returns all the active reminders belonging to the user with id `user_id` in creation order """
+        return list(filter(lambda reminder: reminder.user_id == user_id, self._get_all_reminders()))
+    
+    def _get_unused_reminder_id(self, user_id: int) -> int:
+        """ Returns a reminder id that is not currently in use by the user with id `user_id` """
+        reminders = self._get_user_reminders(user_id)
+        reminder_ids = [reminder.id for reminder in reminders]
+        while (id := random.randint(*REMINDER_IDS_RANGE)) in reminder_ids:
+            continue
+        return id
+
+    def _reached_reminder_limit(self, user: Union[discord.User, discord.Member]) -> bool:
+        """ Returns whether the given user has reached the maximum reminder limit. """
+        # a slight caveat is that if a user sends a command to the bot in DMs, it won't be able to
+        # check for admin perms, so the reminder limit will be enforced
```

Comment:
Theoretically, because the bot is only in one guild, it could query the guild for the user id and check perms, but that's a lot of effort, so... eh

Reply:
Ehhh yeah too much effort :joy: